editor/src/components/canvas/gap-utils.ts

```diff
@@ -0,0 +1,178 @@
+import { MetadataUtils } from '../../core/model/element-metadata-utils'
+import { stripNulls } from '../../core/shared/array-utils'
+import { getLayoutProperty } from '../../core/layout/getLayoutProperty'
+import { defaultEither, isLeft, right } from '../../core/shared/either'
+import { ElementInstanceMetadataMap, isJSXElement } from '../../core/shared/element-template'
+import { canvasRectangle, CanvasRectangle, CanvasVector } from '../../core/shared/math-utils'
+import { optionalMap } from '../../core/shared/optional-utils'
+import { ElementPath } from '../../core/shared/project-file-types'
+import { assertNever } from '../../core/shared/utils'
+import { CSSCursor } from './canvas-types'
+import { CSSNumberWithRenderedValue } from './controls/select-mode/controls-common'
+import { cssNumber, CSSNumber } from '../inspector/common/css-utils'
+
+export type SimpleFlexDirection = 'row' | 'column' | 'row-reverse' | 'column-reverse'
```

Comment:
there's already a SimpleFlexDirection here https://github.com/concrete-utopia/utopia/blob/5c831cb9e4703f3a72175f6aaf7e455b27265109/editor/src/core/layout/layout-utils.ts#L1006
I think we should rename this so it doesn't clash when someone auto-imports it!

Reply:
renamed it to `SimpleFlexDirectionForGap`